core/src/test/java/org/mskcc/cbio/portal/util/TestGenePanelUtil.java

```diff
@@ -0,0 +1,242 @@
+        extractGenes(p, false);
+    }
+
+    @Test
+    public void testExtractGenesMissingTab() {
+        Properties p1 = new Properties();
+        p1.setProperty("gene_list", "ABLIM1 ADAMTS20\tCADM2\tDTNB\tKAT2A\tMSH3\tMYB\tNPIPB15\tOTOR\tP2RY10");
+        Set<CanonicalGene> actual = extractGenes(p1, false);
+        assertNull(actual);
+    }
+
+    @Test
+    public void testExtractGenesNonEmptyGenePanel() {
+        Properties p1 = new Properties();
+        p1.setProperty("gene_list", "ABLIM1\tADAMTS20\tCADM2\tDTNB\tKAT2A\tMSH3\tMYB\tNPIPB15\tOTOR\tP2RY10");
+        Set<CanonicalGene> expected = makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1);
+        Set<CanonicalGene> actual = extractGenes(p1, false);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testExtractGenesAliasedGenes() {
+        Properties p1 = new Properties();
+        p1.setProperty("gene_list", "LIMAB1\tGON-1\tSynCAM 2\tDTNB\tPCAF-b\tDUP\tc-myb\tA-761H5.4\tMIAL1\tLYPSR2");
+        Set<CanonicalGene> expected = makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1);
+        Set<CanonicalGene> actual = extractGenes(p1, false);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testExtractGenesUnknownParseableLong() {
+        String NOT_A_GENE = "111111111111";
+        Properties p1 = new Properties();
+        p1.setProperty("gene_list", NOT_A_GENE + "\tADAMTS20\tCADM2\tDTNB\tKAT2A\tMSH3\tMYB\tNPIPB15\tOTOR\tP2RY10");
+        Set<CanonicalGene> actual = extractGenes(p1, false);
+        assertNull(actual);
+    }
+
+    @Test
+    public void testExtractGenesUnknownUnparseableLong() {
+        String NOT_A_GENE = "HELLOWORLD!";
+        Properties p1 = new Properties();
+        p1.setProperty("gene_list", NOT_A_GENE + "\tADAMTS20\tCADM2\tDTNB\tKAT2A\tMSH3\tMYB\tNPIPB15\tOTOR\tP2RY10");
+        Set<CanonicalGene> actual = extractGenes(p1, false);
+        assertNull(actual);
+    }
+
+    @Test
+    public void testGetAddRemoveAEqualsB() {
+        Pair expected = new Pair(
+            new HashSet<>(),
+            new HashSet<>()
+        );
+        Pair actual = getAddRemove(
+            makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1),
+            makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1)
+        );
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetAddRemoveEmptyIntersection() {
+        Pair expected = new Pair(
+            makeGeneSet(adamts20, ablim1),
+            makeGeneSet(piezo1, fgfr3));
+        Pair actual = getAddRemove(
+            makeGeneSet(adamts20, ablim1),
+            makeGeneSet(piezo1, fgfr3));
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetAddRemoveAOrBEmpty() {
+        Pair expected = new Pair(
+            new HashSet<>(),
```

Comment:
Just to make things _even easier_ to read, I'd do `makeGeneSet()` instead of `new HashSet<>()`

Reply:
for all instances of `new HashSet<>()`?